### DISCUSSION TITLE
Does odoo-helper create a user(odoo) in the system(linux)?

### DISCUSSION THREAD

**Discussion by KomarovAlea** [katyukha/odoo-helper-scripts#17] [General]
I got access to the server where someone had already somehow installed odoo-helper and odoo15. 
For now odoo is running under root, which I really don't like.
Most likely, he install odoo-helper under root, 
because now there is no any users in the /home folder! 
And there are no users in /etc/shadow either. 
And odoo15 project is in the /root folder :-1: 
But Postgres are containe user odoo.

I planning to add server user and make run odoo under them.
I need to use Linux commands to make a server user(odoo)?
or is there a command in odoo-helper to create user odoo?

**Comment by KomarovAlea** [comment]
The funniest thing is that the one who put odoo-helper under root, they did everything according to the instructions!
1. He(she) created a server on Hetzner
2. he(she) open the server terminal (according to Hetzner's instructions!) - i.e. under ROOT!!!
3. he(she) opened the instructions for installing odoo-helper(https://katyukha.gitlab.io/odoo-helper-scripts/installation/), 
but there is not a word about changing the user!
It says: make wget, odoo-helper install...

Here is a log of user commands from history:
```
$ history
1 wget -O - https://gitlab.com/katyukha/odoo-helper-scripts/raw/master/install-system.bash | sudo bash -s
2 odoo-helper install pre-requirements
3 odoo-install -i odoo-15.0 --odoo-version 15.0
4 sudo apt-get install postgresql
5 odoo-helper install postgres odoo odoo
6 cd odoo-15.0
7 odoo-helper restart
```
As you see, it's not command useradd or adduser!

That. the user did everything according to the instructions - and got a problem! 
The user is an ordinary **project manager**! She knows nothing about rights in Linux systems, about root, about different between useradd and adduser and other wisdom of system administration.
You say - user just need to turn on the brain. But I think it is impossible for user.
The problem is that odoo-helper allowed! to install the project under root, in the root folder! 
Odoo-helper did not warn, although it is possible to forbid it by default. And if the user installs with the key --force, then odoo-helper will be allowed.
There is no warning in the documentation, that you cannot run under root. :-(
For example, any sales manager can create projects in odoo.sh! And only a sysadmin who understands the rights of Linux systems can work with Odoo-Helper.
I hope that in the future, Odoo-Helper will become more secure and will not allow an unprepared user to shoot himself in the foot.

**Comment by katyukha** [comment]
Hi,

Thanks for this feedback.

There is note in documentation about production-ready installation. See screenshot below:
![image](https://user-images.githubusercontent.com/1129820/211207125-8da3a748-f0c1-477d-87ac-ad0e7221fc1f.png)

Also, there are a lot of notes in documentation, that describes this project as intendend primarily for developers, to simplify local development process.

For installation of Odoo on servers, there is separate project [CRND Deploy](https://github.com/crnd-inc/crnd-deploy) that is designed to install odoo on production servers with proper access rights and preconfigured nginx. That project, installs and configures odoo-helper and odoo in right way to work as single odoo instance on server. Also, that script generates init file, to ensure that odoo server will be automatically started after reboot of physical machine.

Possibly, the documentation could be updated to state more clearly, that another project have to be used to install Odoo on production servers.

---

The fix that adds safeguard [was added](https://gitlab.com/katyukha/odoo-helper-scripts/-/merge_requests/37/diffs?commit_id=6cf2c13ca4e6fd7f66bb38b6f8e06e8717e89d18) to dev branch, and will be available in next release.

---

PS. If you have any ideas have to improve documentation, pull/merge requests are welcome.
